### ISSUE TITLE
Uncaught (in promise) TypeError: Cannot read properties of null (reading '__H')

### ISSUE THREAD

**Issue by philosofonusus** [preactjs/preact-cli#1701]
- [x ] Check if updating to the latest Preact version resolves the issue

**Describe the bug**
The bug happens when I open compiled application in browser, I see white screen and error in console

**To Reproduce**
```
import React, { render } from 'preact/compat';
import {h} from 'preact';
import * as Sentry from '@sentry/react';
import { BrowserTracing } from '@sentry/tracing';
import App from './components/app';
import 'normalize.css';

if (typeof window !== 'undefined') {
  render(h(App, {}), document.body);
}

```

run ```preact build --no-prerender```
run ```npm run serve```
open application in browser
see white screen
see error in console

**Expected behavior**
What should have happened when following the steps above?
Application should run

**Comment by rschristian** [comment]
Can't reproduce with those instructions.

The most I see in the console is the graphql failing, which is to be expected. Your issue might occur only after, or on a different page?

Ideally we'd want a much more minimal reproduction, as there's quite a bit going on in that app.

**Comment by philosofonusus** [comment]
> Can't reproduce with those instructions.
> 
> The most I see in the console is the graphql failing, which is to be expected. Your issue might occur only after, or on a different page?
> 
> Ideally we'd want a much more minimal reproduction, as there's quite a bit going on in that app.

you may expirirence graphql failing  because you start without --no-prerender

**Comment by rschristian** [comment]
> you may expirirence graphql failing because you start without --no-prerender

No, the graphql fails because I don't have your server to query against.

I'm able to build just fine, the browser console just reports the graphql issue. Nothing related to Preact.

**Comment by philosofonusus** [comment]
> > you may expirirence graphql failing because you start without --no-prerender
> 
> No, the graphql fails because I don't have your server to query against (`sirv` ends up responding with the HTML doc).
> 
> I'm able to build just fine, the browser console just reports the graphql issue. Nothing related to Preact.
This is what I see ![Screenshot from 2022-06-17 18-59-27](https://user-images.githubusercontent.com/32138582/174334546-40f27aae-e198-4180-bf49-53984724829b.png)

Do you see the main page of the application, after you bundle and serve the application? What version of preact-cli do you use? 
```
npm i && preact build --no-prerender && npm run serve
``

**Comment by philosofonusus** [comment]
Have tried downgrading to preact-cli@3.0.0-rc.18 still doesnt work

**Comment by rschristian** [comment]
> Do you see the main page of the application, after you do `npm i && preact build --no-prerender && npm run serve`?

![temp](https://user-images.githubusercontent.com/33403762/174336639-e2a909ca-77d8-4a0f-99b4-292a8f513795.png)

Loads just fine, besides the gql error.

> What version of preact-cli do you use?

NPM installs according to the lock file that you distributed. Seems to be `3.3.5`.

> Have tried downgrading to preact-cli@3.0.0-rc.18 still doesnt work

Definitely don't downgrade to a years-old version.

**Comment by philosofonusus** [comment]
> > Do you see the main page of the application, after you do `npm i && preact build --no-prerender && npm run serve`?
> 
> ![temp](https://user-images.githubusercontent.com/33403762/174336639-e2a909ca-77d8-4a0f-99b4-292a8f513795.png)
> 
> Loads just fine, besides the gql error.
> 
> Edit: Whoops, my personal port preference (`3000`) lead to that specific gql error (it should 404 instead). Still, the page loads just fine.
> 
> > What version of preact-cli do you use?
> 
> NPM installs according to the lock file that you distributed. Seems to be `3.3.5`.
> 
> > Have tried downgrading to preact-cli@3.0.0-rc.18 still doesnt work
> 
> Definitely don't downgrade to a years-old version.

According to your screenshot everything works as expected but still i don't understand why I see this error.

**Comment by rschristian** [comment]
Without a reproduction I can only guess.

Usually that error means Preact is duplicated in your bundles. It could be your use of `preact/compat` and `preact` in your entrypoint?

```diff
-import React, { render } from 'preact/compat';
import {h} from 'preact';
import * as Sentry from '@sentry/react';
import { BrowserTracing } from '@sentry/tracing';
import App from './components/app';
import 'normalize.css';

if (typeof window !== 'undefined') {
  //   Sentry.init({
  //     dsn: "https://31b172cca34f44a1a41f244fd6fa4f50@o1131806.ingest.sentry.io/6176685",
  //     integrations: [new BrowserTracing()],
  //     tracesSampleRate: 1.0,
  //   });

-  render(h(App, {}), document.body);
}
+export default App;
```

CLI will handle rendering for you. The window check is also unnecessary if you're not prerendering your app.

**Comment by philosofonusus** [comment]
> Without a reproduction I can only guess.
> 
> Usually that error means Preact is duplicated in your bundles. It could be your use of `preact/compat` and `preact` in your entrypoint?
> 
> ```diff
> -import React, { render } from 'preact/compat';
> import {h} from 'preact';
> import * as Sentry from '@sentry/react';
> import { BrowserTracing } from '@sentry/tracing';
> import App from './components/app';
> import 'normalize.css';
> 
> if (typeof window !== 'undefined') {
>   //   Sentry.init({
>   //     dsn: "https://31b172cca34f44a1a41f244fd6fa4f50@o1131806.ingest.sentry.io/6176685",
>   //     integrations: [new BrowserTracing()],
>   //     tracesSampleRate: 1.0,
>   //   });
> 
> -  render(h(App, {}), document.body);
> }
> +export default App;
> ```
> 
> CLI will handle rendering for you. The window check is also unnecessary if you're not prerendering your app.

If I do that, then I get this in dev run 
![Screenshot from 2022-06-17 19-44-23](https://user-images.githubusercontent.com/32138582/174341986-43838492-dc15-4074-b65c-2f7ba3cb6627.png)
And even if I build it I still get this issue
```
Uncaught (in promise) TypeError: Cannot read properties of null (reading '__H')
```

**Comment by philosofonusus** [comment]
I'll try to build it not on local machine and see what happens

**Comment by rschristian** [comment]
Hmm, well mixing up `preact`, `preact/hooks`, and `preact/compat` doesn't necessarily fill me with confidence (generally you don't want to be importing from `preact/compat`), though I'm not sure where that error is coming from. Way too much there for me to spend time digging through to find the source. I suppose that's a separate issue that you can ignore for now if your workaround hides it.

Let me know if you can reproduce the error more minimally, happy to look into it. Copy/pasting your app into codesandbox or something and deleting routes/components until you can no longer reproduce might be a decent way of narrowing it down.

**Comment by rschristian** [comment]
Closing, as I cannot reproduce.

If you can get a minimal reproduction of the error together I'd be happy to take a look.
